horace_core/sqw/PixelData/@PixelDataFileBacked/do_unary_op.m

```diff
@@ -9,7 +9,9 @@
 
 pix_out = obj;
 
-pix_out = pix_out.get_new_handle();
+if isempty(pix_out.file_handle_)
+    pix_out = pix_out.get_new_handle();
+end
```

Comment:
I suspect this is a bug or unacceptably inefficient code. 
page averages have to be calculated per page and then combined together. 
In addition, signal and error of the DnD image have to be calculated in the same loop as pixels (they are also "kind of" pixel averages. 

If everything here formally works (I may assume that you loop over pixels here) then loop over pixels calculateing pixel averages, and then loop over pixels on SQW level recalculating DnD.s, DnD.e, and DnD.npix (probably untached here, unless you are masking) this unacceptable.

Looping over avearage file on iDaaaS now takes about 5-10 minutes at best. Doing it three times would not satisfy anybody.

Reply:
As noted, performance is not a concern at this stage. Functionality is, we can optimise it later. As it is, this is an infinite speedup over whatever exists for cases where file backing is necessary.